src/zap/zapimage.cpp

```diff
@@ -627,6 +627,13 @@ void ZapImage::AllocateVirtualSections()
         // Debug info is sometimes used during exception handling to build stacktrace
         //
         m_pDebugSection = NewVirtualSection(pTextSection, IBCUnProfiledSection | ColdRange | DebugSection, sizeof(DWORD));
+
+#ifdef FEATURE_READYTORUN_COMPILER
+        if (IsReadyToRunCompilation())
+        {
+            m_pAvailableTypesSection = NewVirtualSection(pTextSection, IBCUnProfiledSection | HotRange | ReadonlySection);
+        }
+#endif    
     }
```

Comment:
The virtual sections are meant to be created from the hottest to the coldest, to improve locality of the image. Since this section is relatively warm, it should be created sooner - I would put it around m_pCodeMethodDescsSection above.